### PR TITLE
Fix getUnique to always return allocated array

### PR DESCRIPTION
**Pull request by aarograh**
Ensures that an allocated array is always returned by getUnique

### CODE REVIEW

**Comment by aarograh** [comment]
@djabaay this is related to https://code.ornl.gov/vera/vera-dev/-/issues/4634 that I just tagged you in.  Mind doing a quick review?  It's just a quick change, but I think ensuring the return value is always allocated, even if to size 0, is a little safer than before.

**Comment by djabaay** [comment]
I guess if you want the routine to be able to accept an unallocated array to execute on, unallocated or allocated to size 0 shouldn't really matter.  Is the goal to just ensure that _something_ is allocated, regardless of the input?

I haven't got my ucams unlocked so I haven't seen the ticket yet.

**Comment by aarograh** [comment]
Yeah I think it's better to return something allocated no matter what.  To say there are 0 unique entries in a size-0 or unallocated array seems like a reasonable statement to me.

The issue this is aiming to address is a crash in BWR jumpin cases that Mehdi is dealing with.

**Comment by djabaay** [comment]
Understood, thanks for the brief explanation.

File: src/ArrayUtils.f90 [hunk 1]
```diff
@@ -524,6 +524,8 @@ SUBROUTINE getUnique_1DReal(r,rout,delta,tol)
 
     !Deallocate
     DEALLOCATE(tmpr)
+  ELSE
+    ALLOCATE(rout(0))
```
Comment (djabaay) [inline src/ArrayUtils.f90:528]: The only thing I can think of is add a comment in the subroutine header to describe when there are 0 unique entries.  Other than comments, looks good.

**Review by djabaay — CHANGES_REQUESTED** [review]

File: src/ArrayUtils.f90 (hunk 1 above)
Comment (aarograh) [reply src/ArrayUtils.f90:529]: good call.  Just pushed updates

**Review by djabaay — APPROVED** [review]
Thanks @aarograh

**Pull request merged**